core/services/ocr2/plugins/ocr2keeper/util.go

```diff
@@ -72,7 +72,7 @@ func EVMDependencies(spec job.Job, db *sqlx.DB, lggr logger.Logger, set evm.Chai
 	}
 
 	rAddr := ethkey.MustEIP55Address(oSpec.ContractID).Address()
-	if registry, err = kevm.NewEVMRegistryServiceV2_0(rAddr, chain, mc, lggr); err != nil {
+	if registry, err = kevm.NewEVMRegistryServiceV2_1(rAddr, chain, mc, lggr); err != nil {
```

Comment:
How is the clash between this file and `core/services/ocr2/plugins/ocr2keeper20/util.go` avoided? I see they are in the same package and have same function names. 

These functions are called in the top level delegate https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/delegate.go#L649-L659 

would be good to plug this 2.1 change within delegate too within this PR to clarify how we choose 2.0 vs 2.1

Reply:
A ContractVersion field will be added to job spec once 2.1 is stable, with default value of 2.0 to support older job specs